### PR TITLE
Package superlu_dist

### PR DESCRIPTION
**Pull request by davidbrochart**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by davidbrochart** [comment]
I'm not sure with it fails on OSX and not on Linux. It seems that the dependencies are not compatible.

File: recipes/superlu_dist/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - parmetis 4.0.3*
+    - metis 5.1.0*
+
+test:
+  commands:
+    - test -f ${PREFIX}/include/superlu_defs.h  # [unix]
+
+about:
+  home: http://crd-legacy.lbl.gov/~xiaoye/SuperLU/
+  license: University of California
```
Comment (SylvainCorlay) [inline recipes/superlu_dist/meta.yaml]: Is this the actual license name?

File: recipes/superlu_dist/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('fortran') }}
+    - cmake >=2.8.12
+    - gcc  # [osx]
+    - openblas  # [unix]
+    - parmetis 4.0.3*
+    - metis 5.1.0*
```
Comment (SylvainCorlay) [inline recipes/superlu_dist/meta.yaml]: I think openblas, parmetis and metis should only be host dependencies.

File: recipes/superlu_dist/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  host:
+    - parmetis 4.0.3*
+    - metis 5.1.0*
+    - openmpi
+  run:
+    - libgcc  # [osx]
```
Comment (SylvainCorlay) [inline recipes/superlu_dist/meta.yaml]: libgcc should normally be a strong run_exports of gcc, so I don't think you need to mention it.

File: recipes/superlu_dist/meta.yaml (hunk 1 above)
Comment (davidbrochart) [reply recipes/superlu_dist/meta.yaml]: The full name is [Copyright (c) 2003, The Regents of the University of California, through
Lawrence Berkeley National Laboratory (subject to receipt of any required 
approvals from U.S. Dept. of Energy)](https://portal.nersc.gov/project/sparse/superlu/License.txt).

**Pull request closed without merging**